README.md

```diff
@@ -1,2 +1,213 @@
+
+- Create a GitHub/GitLab repository for each Practical if one doesn't already exist.
+
+- Learn Basic DevOps Principles:
+  - Learn how to use Version Control effectively for your codebase as Git is Fundamental.
+  - Know how to write unit tests.
+  - Start by understanding the core DevOps concepts like continuous integration, continuous deployment, and automation.
+  - Learn some basic scripts to get a basic understanding of scripting language.
+  - Be aware of basic security practices like exposing env vars, Inbound/Outbound security rules, etc…
+
+- Some Basic Guidelines for CI/CD:
+  - Understand the CI/CD concepts and their significance with pipeline usage.
+  - Learn how to set up the basic CI/CD pipeline with the GitLab/GitHub-CI tool.
+  - All lint rules should be passed from the pipeline.
+  - Always use the latest versions of languages and project dependencies
+  - Understand how to manage project dependencies using Node modules, Composer, or Vendors. And integrate them into the pipeline.
+  - Learn to automate the testing into the CI/CD pipeline.
+  - Learn how to use environment variables effectively in CI/CD pipelines including handling sensitive information like secret keys or passwords.
+
+# Prerequisites
+
+Learn following before starting practice.
+
+- Linux commands for DevOps
+    - Install dependencies
+    - File operations
+    - SSH commands
+- [Docker](https://docs.docker.com/get-started/overview/)
+    - Concepts
+      - swarm
+      - services
+      - images
+      - containers
+      - tags
+      - volume
+      - docker-compose
+    -  Create image
+    -  Create container
+    -  Run container locally
+    -  Push on the registry
+    -  Remove unused services or data
+- [Version control system](https://learn.microsoft.com/en-us/devops/develop/git/what-is-version-control) and [git](https://www.nobledesktop.com/learn/git/what-is-git)
+- [What is cloud computing?](https://aws.amazon.com/what-is-cloud-computing/)
+
+---
+
+# Ensure code quality
+
+### Practical 1 - Linting javascript code with eslint
+
+**Expected -** The GitLab workflow job for Linting and Formatting should be passed without errors.
+
+- Set up the Vue.js project with typescript and create the simple component with tailwind css.
+  
+- Integrate eslint and perform following rules,
+  - Enforce naming conventions for files and folder names
+  - Enforce naming conventions for variables and function names
+  - Use semicolons at the end of statements
+  - Use double quotes for strings
+  - Enforce no trailing spaces, consistent line endings (e.g., LF or CRLF), and consistent spacing around operators
+  - Limit the line length to a maximum of 100 characters
+  - Limit the max 300 lines for files
+  - Limit the max 30 lines for functions
+  - Enforce 2 space identation
+  - Ignore eslint check in non-javascript files
+  
+- Write GitLab CI script to run lint
```

Comment:
Instead of suggesting either one from GitLab and GitHub, including both will help to differentiate process closely.

Reply:
I think it will take more time to configure things on both platforms. We have added both alternatively, so the developer can have experience with both.